typespec-extension/src/code-model-builder.ts

```diff
@@ -1109,68 +1066,68 @@ export class CodeModelBuilder {
       // format if array
       let style = undefined;
       let explode = undefined;
-      if (param.param.type.kind === "Model" && isArrayModelType(this.program, param.param.type)) {
-        if (param.type === "query") {
-          // eslint-disable-next-line deprecation/deprecation
-          const queryParamFormat = param?.format;
-          if (queryParamFormat) {
-            switch (queryParamFormat) {
-              case "csv":
-                style = SerializationStyle.Simple;
-                break;
-
-              case "ssv":
-                style = SerializationStyle.SpaceDelimited;
-                break;
-
-              case "tsv":
-                style = SerializationStyle.TabDelimited;
-                break;
-
-              case "pipes":
-                style = SerializationStyle.PipeDelimited;
-                break;
-
-              case "multi":
-                style = SerializationStyle.Form;
-                explode = true;
-                break;
-
-              default:
-                this.logWarning(`Unrecognized query parameter format: '${queryParamFormat}'.`);
-                break;
-            }
+      if (sdkType.kind === "array") {
+        if (param.kind === "query") {
+          const format = param.collectionFormat;
+          switch (format) {
+            case "csv":
+              style = SerializationStyle.Simple;
+              break;
+
+            case "ssv":
+              style = SerializationStyle.SpaceDelimited;
+              break;
+
+            case "tsv":
+              style = SerializationStyle.TabDelimited;
+              break;
+
+            case "pipes":
+              style = SerializationStyle.PipeDelimited;
+              break;
+
+            case "multi":
+              style = SerializationStyle.Form;
+              explode = true;
+              break;
+
+            default:
+              if (format) {
+                this.logWarning(`Unrecognized query parameter format: '${format}'.`);
+              }
+              break;
           }
```

Comment:
I think Chenjie has explained all format. https://github.com/Azure/typespec-azure/pull/1363#discussion_r1716327246

Please handle them, e.g. simple and form